wasm/jit/jit_arm64.go

```diff
@@ -327,16 +356,87 @@ func (c *arm64Compiler) compileCallIndirect(o *wazeroir.OperationCallIndirect) e
 	return fmt.Errorf("TODO: unsupported on arm64")
 }
 
+// compileDrop implements compiler.compileDrop for the arm64 architecture.
 func (c *arm64Compiler) compileDrop(o *wazeroir.OperationDrop) error {
-	return fmt.Errorf("TODO: unsupported on arm64")
+	r := o.Range
+	if r == nil {
+		return nil
+	} else if r.Start == 0 {
+		// If the drop starts from the top, we just mark all the regiters
+		// used by drop targets unused.
+		for i := 0; i <= r.End; i++ {
+			if loc := c.locationStack.pop(); loc.onRegister() {
+				c.markRegisterUnused(loc.register)
+			}
+		}
+		return nil
+	}
+
+	// Save the non-drop target values.
+	liveValues := make([]*valueLocation, r.Start)
+	for i := 0; i < r.Start; i++ {
+		live := c.locationStack.pop()
+		liveValues[i] = live
+	}
+
+	// Then mark all registers used by drop tragets unused.
```

Comment:
s/tragets/targets/g and mention why `r.End-r.Start+1` or extract that into a local which name says why